src/build_workflow/builder_from_source.py

```diff
@@ -39,7 +44,7 @@ def build(self, build_recorder: BuildRecorder) -> None:
                     f"-v {self.target.version}",
                     f"-p {self.target.platform}",
                     f"-a {self.target.architecture}",
-                    f"-d {self.target.distribution}" if self.target.distribution else None,
+                    f"-d {self.target.distribution}" if self.target.distribution and (self.component.name in DISTRIBUTION_SUPPORTED_COMPONENTS) else None,
```

Comment:
Checking the name will break silently. 

Is the component type here `ComponentOpenSearchMin` or `ComponentOpenSearchDashboardsMin`? I believe it is, and in that case at least check whether `type(self.component) in [ComponentOpenSearchMin, ComponentOpenSearchDashboardsMin]`.

Reply:
Could you elaborate more on this? Why do we need to check component type?

Thanks.

Reply:
The instance of `component` is not `Component`, it's something that's a child of that class, either `ComponentOpenSearchMin` or `ComponentOpenSearchDashboardsMin` therefore you don't need to check whether `.name == "OpenSearch"`, that's already done elsewhere, where we needed to know whether it's one or the other. Checking the name here would duplicate that logic.

Reply:
Actually looking at the code it looks like we only do that in the maniests workflow :(

I'll approve this hack for now and we can refactor later.

Reply:
After checking, it is actually `manifests.input_manifest.InputComponentFromSource` object and the name is directly taking from github repo name.

```
class InputComponentFromSource(InputComponent):
    def __init__(self, data: Any) -> None:
        super().__init__(data)
        self.repository = data["repository"]
        self.ref = data["ref"]
        self.working_directory = data.get("working_directory", None)

    def __stabilize__(self) -> None:
        ref, name = GitRepository.stable_ref(self.repository, self.ref)
        logging.info(f"Updating ref for {self.repository} from {self.ref} to {ref} ({name})")
        self.ref = ref

    def __to_dict__(self) -> dict:
        return {
            "name": self.name,
            "repository": self.repository,
            "ref": self.ref,
            "working_directory": self.working_directory,
            "checks": list(map(lambda check: check.__to_dict__(), self.checks)),
            "platforms": self.platforms,
        }
```

Will use this method for now.